### DISCUSSION TITLE
Streaming two separate livestreams in Bonsai

### DISCUSSION THREAD

**Discussion by jackiekuyat** [bonsai-rx/bonsai#1997] [Q&A] [answered]
I could use help figuring out how to livestream two separate videos using Bonsai. I can get the video streams to pop up, but as you can see in the screenshots, one of the video streams shows up as a white box, while the other works fine. The same happens if I have two instances of Bonsai open and run them one at a time.   

The only code running is the Bonsai workflow at the front of the screenshot. 

![Screenshot 2024-09-04 104949](https://github.com/user-attachments/assets/304d9ab0-d92e-4832-950e-f5cefc563f42)
![Screenshot 2024-09-04 104956](https://github.com/user-attachments/assets/d573017a-7ed9-452f-b52d-210c9ae157b3)

**Comment by bruno-f-cruz** [comment]
Does it work if you only connect to the camera that generates the white window? Are the two cameras coupled in hardware somehow?

**Reply by jackiekuyat** [reply]
Yes, I can get each camera to stream separately if I disable the other camera. 

Both cameras are connected to the same computer via different USB 3.0s. 

![Screenshot 2024-09-04 115028](https://github.com/user-attachments/assets/6f5b8e8b-c7f3-4904-a44e-896da2c20ffc)
![Screenshot 2024-09-04 115057](https://github.com/user-attachments/assets/22333257-78ec-4532-9613-4e4a0ed2f8e4)

![1000000852](https://github.com/user-attachments/assets/61f1eab3-a472-4d4a-8be6-f7c7f54f6c53)

**Reply by jackiekuyat** [reply]
Is Bonsai not made to stream more than one distinct camera at a time? I know it can handle two cameras if I want to sync them, but in my case, I don't want to do this as I am trying to stream video from two different behavior boxes. 

I also run into the same camera streaming issue if I have two instances of Bonsai open. So one workspace controls the bye camera and the other workspace controls the Hi camera.

**Reply by bruno-f-cruz** [reply]
I have not encountered problems acquiring from different cameras at the same time before so i am puzzled. have you tried different camera models? Maybe use CameraCapture instead of VideoCaptureDevice?

**Reply by jackiekuyat** [reply]
Using CameraCaputure causes "Unable to acquire camera frame" error to occur: 
![Screenshot 2024-09-05 092224](https://github.com/user-attachments/assets/7055a34c-efce-4f15-a447-f977870cf5bd)
![Screenshot 2024-09-05 092301](https://github.com/user-attachments/assets/9861cffc-c159-449d-8145-d229793f3a8d)

The video feed popouts appear briefly before the error occurs.

**Reply by bruno-f-cruz** [reply]
Can you add the workflow?

**Reply by jackiekuyat** [reply]
Apologies I am new to Bonsai so I am going to ask a silly clarifying question. Isn't the workflow the page I took a screenshot of?

**Reply by bruno-f-cruz** [reply]
The .bonsai file is the workflow

**Reply by jackiekuyat** [reply]
Oh! Here it is: 
[Bonsai.zip](https://github.com/user-attachments/files/16897674/Bonsai.zip)

**Reply by bruno-f-cruz** [reply, marked as answer]
With my current setup (2 cameras as well) that workflow acquires fine. So i really think it may be a setup configuration issue. Try on another PC / different cameras?

**Reply by jackiekuyat** [reply]
Thanks for testing the workflow, you were right. I connected the camera's to a different computer and it worked fine. Thank you for your help!
